no-vary-search.md

```diff
@@ -0,0 +1,340 @@
+
+### Carrying data that is or can be processed by client-side script only
+
+Some examples:
+
+- Analytics data, e.g. what marketing campaign led the user to the destination.
+- Where the user should be redirected to after some action is complete, e.g. a login process.
+- Some detail to highlight on the page, e.g. map coordinates to zoom to or a product variant to emphasize.
+
+Although in theory such data could be passed via the fragment instead of query string, prevailing practice often uses the query string. Additionally, preloading changes the calculus here: a page might want to do server-side processing if possible, but client-side processing if doing so would unlock the fast-loading benefits of preloading.
+
+_Note: analytics, where you carry generic data such as the marketing campaign or referring source, is different than [navigational tracking](https://privacycg.github.io/nav-tracking-mitigations/#terminology), which passes along user identifiers. Nothing about this proposal is intended to aid navigational tracking, and we expect it to be fully compatible with future and [currently-deployed mitigations](https://privacycg.github.io/nav-tracking-mitigations/#deployed-mitigations) for navigational tracking. See more discussion in our [privacy considerations](#security-and-privacy-considerations)._
+
+Another use case for this is entirely client-side-rendered skeleton pages. However, we're not focused on such cases, since they often use the path component to carry the variable information. See [future work on `No-Vary-Path`](#no-vary-path) for more discussion.
+
+### Carrying data not yet determined at the time of preloading
+
+The most prominent example here is analytics. Consider:
+
+```html
+<script type="speculationrules">
+{
+  "prerender": [{
+    "source": "list",
+    "urls": ["/articles/new-underwater-phone"]
+  }]
+}
+</script>
+
+<a href="/articles/new-underwater-phone?via=heroimage">
+  <img src="underwaterphone.jpg" alt="A phone, underwater!">
+</a>
+
+<a href="/articles/new-underwater-phone?via=headline">New underwater phone, just released!</a>
+```
+
+When the page loads, we don't know yet whether the user will click on the hero image or the headline link. So it's useful to prerender the page with no query parameter, and then later [communicate that information](#prerendering-activation) once the user clicks on a link.
+
+## Detailed design
+
+### The header
+
+The `No-Vary-Search` header is a [HTTP structured field](https://www.rfc-editor.org/rfc/rfc8941.html) whose value must be a list. The list can contain either predefined [tokens](https://www.rfc-editor.org/rfc/rfc8941.html#name-tokens), which have special behavior, or [strings](https://www.rfc-editor.org/rfc/rfc8941.html#name-strings), which indicate query parameters not to vary on. The special tokens are:
+
+- `order`: indicates the order of the query parameters should be ignored when constructing the cache key.
```

Comment:
Do we want to clarify that we mean only relative to distinct keys? (At least, that is what https://url.spec.whatwg.org/#dom-urlsearchparams-sort does and seems likely to be the most useful.)